src/lib/manager/cloudFrontManager.ts

```diff
@@ -139,6 +141,14 @@ export class CloudFrontManager {
         cachePolicy: props.defaultBehavior ? props.defaultBehavior.cachePolicy : undefined,
         origin: origin,
         originRequestPolicy: props.defaultBehavior ? props.defaultBehavior.originRequestPolicy : undefined,
+        functionAssociations: cloudfrontFunction
```

Comment:
This will always default any associated functions to `Viewer Request` event type and doesn't allow multiple. 
...Instead the whole array has to be passed as a parameter => `functionAssociations: props.functionAssociations`
... and you will have to build the input array separately and pass it through